src/mediarithmics/api/datamart/UserActivityInterface.ts

```diff
@@ -0,0 +1,169 @@
+    [propsName: string]: any;
+}
+
+export interface UserVisitActivity extends UserActivity {
+    $session_duration?: number;
+    $topics?: Map<string, Map<string, number>>;
+    $site_id?: string;
+    $app_id?: string;
+}
+
+export interface UserActivityOrigin
+    extends Customizable{
+    $campaign_id?: number;
+    $campaign_name?: string;
+    $channel?: string;
+    $creative_id?: number;
+    $creative_name?: string;
+    $engagement_content_id?: string;
+    $gclid?: string;
+    $keywords?: string;
+    $log_id?: string;
+    $message_id?: number;
+    $message_technical_name?: string;
+    $referral_path?: string;
+    $social_network?: string;
+    $source?: string;
+    $sub_campaign_id?: number;
+    $sub_campaign_technical_name?: string;
+    $ts?: number;
+}
+
+export interface UserActivityLocation {
+    $source?: LocationSourceEnum;
+    $country?: string;
+    $region?: string;
+    $iso_region?: string;
+    $city?: string;
+    $iso_city?: string;
+    $zip_code?: string;
+    $latlon: number[];
+
+    [propsName: string]: any;
+}
+
+
+export interface UserActivityEventProperty {
+    [propsName: string]: any;
+}
+
+export interface CampaignTrackingProperties
+    extends Customizable {
+    $campaign_technical_name?: string;
+    $sub_campaign_technical_name?: string;
+    $message_technical_name?: string;
+    $creative_technical_name?: string;
+    $campaign_id?: number;
+    $sub_campaign_id?: number;
+    $message_id?: number;
+    $creative_id?: number;
+}
+
+export interface ConversionProperties
+    extends Customizable {
+    $conversion_id?: string;
+    $goal_id?: number;
+    //@Deprecated
+    $conversion_technical_id?: string;
+    //@Deprecated
+    $goal_technical_id?: string;
+    $conversion_value?: number;
+    $log_id?: string;
+    $conversion_external_id?: string;
+    $goal_technical_name?: string;
+}
+
+export type EventName =
+    '$ad_click'
+    | '$ad_view'
+    | '$conversion'
+    | '$category_view'
+    | '$page_view'
+    | '$home_view'
+    | '$item_view'
+    | '$item_list_view'
+    | '$product_view'
+    | '$product_list_view'
+    | '$basket_view'
+    | '$transaction_confirmed'
+    | '$email_view'
+    | '$email_click'
+    | '$email_sent'
+    | '$email_delivered'
+    | '$email_soft_bounce'
+    | '$email_hard_bounce'
+    | '$email_unsubscribe'
+    | '$email_complaint'
+    | '$set_user_profile_properties'
+    | '$content_corrections'
+    ;
+
+
+export type UserActivityEvent =
+    AdClickEvent
+    | AdViewEvent
+    | ConversionEvent
+    | GenericUserActivityEvent
+    ;
+
+export interface AdClickEvent {
+    $ts: number;
+    $event_name: '$ad_click';
+    $properties: CampaignTrackingProperties
+}
+
+export interface AdViewEvent {
+    $ts: number;
+    $event_name: '$ad_view';
+    $properties: CampaignTrackingProperties
+}
+
+export interface ConversionEvent {
+    $ts: number;
+    $event_name: '$conversion';
+    $properties: ConversionProperties
+}
+
+//TOFIX
```

Comment:
From what I remember, `GenericUserActivityEvent` is kind of a wildcard and we should rather complete the definition of `UserActivityEvent` so that `GenericUserActivityEvent` becomes useless